lib/elixir/test/elixir/module_test.exs

```diff
@@ -164,6 +164,12 @@ defmodule ModuleTest do
     module = Very.Long.Module.Name.And.Even.Longer
     assert Module.split(module) == ["Very", "Long", "Module", "Name", "And", "Even", "Longer"]
     assert Module.split("Elixir.Very.Long") == ["Very", "Long"]
+    assert_raise FunctionClauseError, fn ->
+      Module.split(:just_an_atom) # #3161 regression
```

Comment:
We can drop the comments here, just remember to mention it in the commit message. :)